graph-data.rs/src/check_signatures.rs

```diff
@@ -0,0 +1,127 @@
+use anyhow::Result as Fallible;
+use anyhow::{format_err, Context};
+use futures::stream::{FuturesOrdered, StreamExt};
+use lazy_static::lazy_static;
+use reqwest::{Client, ClientBuilder};
+use semver::Version;
+use std::collections::HashSet;
+use std::ops::Range;
+use std::str::FromStr;
+use std::time::Duration;
+use url::Url;
+
+use cincinnati::plugins::prelude_plugin_impl::TryFutureExt;
+use cincinnati::Release;
+// base url for signature storage - see https://github.com/openshift/cluster-update-keys/blob/master/stores/store-openshift-official-release-mirror
+lazy_static! {
+  static ref BASE_URL: Url =
+    Url::parse("https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/")
+      .expect("could not parse url");
+}
+
+static DEFAULT_TIMEOUT_SECS: u64 = 30;
+
+// CVO has maxSignatureSearch = 10 in pkg/verify/verify.go
+static MAX_SIGNATURES: u64 = 10;
+
+fn payload_from_release(release: &Release) -> Fallible<String> {
+  match release {
+    Release::Concrete(c) => Ok(c.payload.clone()),
+    _ => Err(format_err!("not a concrete release")),
+  }
+}
+
+async fn fetch_url(client: &Client, sha: &str, i: u64) -> Fallible<()> {
+  let url = BASE_URL
+    .join(format!("sha256={}/", sha).as_str())?
```

Comment:
nit: this assumes that we will stick with sha256.  Safe enough in the near term, but it would be more robust to have this take the full digest and [replace the `:` with `=`][1].

[1]: https://github.com/containers/image/blame/f1a6f9b50d0286bf208935d4b9ab48117294ac31/docs/signature-protocols.md#L39-L51